Gems/Atom/Utils/Code/Include/Atom/Utils/ImGuiCpuProfiler.inl

```diff
@@ -43,6 +45,22 @@ namespace AZ
                 return AZStd::string::format("Thread: %zu", static_cast<size_t>(threadId));
             }
 
+            inline AZStd::thread_id ThreadIdFromNumeric(uintptr_t deserializedThreadId)
+            {
+                if constexpr (AZStd::is_pointer_v<AZStd::native_thread_id_type>)
+                {
+                    // native_thread_id_type and uintptr_t are implicitly convertible, pass in directly.
+                    return AZStd::thread_id(deserializedThreadId);
+                }
+                else
+                {
+                    // sizeof(native_thread_id_type) <= sizeof(uintptr_t), so we need to introduce a lossy cast.
+                    // We accept data loss here because the actual thread_id does not matter, we just need deserialized
+                    // IDs to uniquely group together regions that executed on the same thread.
+                    return AZStd::thread_id(azlossy_cast<AZStd::native_thread_id_type>(deserializedThreadId));
+                }
```

Comment:
Note: this is really the critical section of this code - I think that this should not break the mac build but I don't have any real way to test it. Additionally, it feels a little dirty to skirt around the PAL like this, but I'd argue the complexity of going and PALifying this would be less maintainable than the `if constexpr` approach. Thoughts?

Reply:
I agree, I don't think it's worth PAL-ifying.

Reply:
Taking a second look at this, I think the current code will not work on Mac. We can replicate what would happen if `AZStd::native_thread_id_type` is a pointer by checking this assert which fails for me:
```cpp 
static_assert(AZStd::is_convertible_v<uintptr_t, void*>);
```
which means that we cannot implicitly convert the integral type to a pointer. Instead we need to `reinterpret_cast` since this is an unsafe conversion, but adding the cast makes the build fail since it looks like if-constexpr still compiles the branch. Going to investigate using templates but may have to go with pal.

Reply:
Also, it looks like adding a cstyle cast to `native_thread_id_type` in the first if branch works (and I believe would fix Mac as well) but would be pretty ugly.